src/wms/src/main/java/org/geoserver/wms/GetMap.java

```diff
@@ -669,7 +669,24 @@ WebMap executeInternal(
                     }
                 }
                 if (!merged) {
-                    WMSLayer Layer = new WMSLayer(wms, gt2Layer);
+                    WMSLayer Layer = null;
+                    // GEOS-9312
+                    if (mapLayerInfo.getLayerInfo() instanceof CascadedLayerInfo) {
+                        CascadedLayerInfo cascadedLayerInfo =
+                                (CascadedLayerInfo) mapLayerInfo.getLayerInfo();
+
+                        String style = request.getStyles().get(i).getName();
+                        String imageFormat = request.getFormat();
+                        // if passed style does not exist in remote, default to forced
+                        if (!cascadedLayerInfo.isSelectedRemoteStyles(style))
```

Comment:
If the passed style doesn't exists it shoudl fail, I mean right now if send a WMS requets to GeoServer with some invalid style I expect the request to fail, otherwise how do I know that requested style doesn't actually exists?

Reply:
understood, if it goes not exist, we throw an exception

Reply:
kk